Tasks/app-store-promote/task.json

```diff
@@ -46,6 +47,42 @@
             "helpMarkDown": "The TFS/Azure DevOps service connection that specifies the identity used to publish the app.",
             "visibleRule": "authType = ServiceEndpoint"
         },
+        {
+            "name": "apiKeyId",
+            "type": "string",
+            "label": "App Store Connect API Key ID",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "The key_id of the account used to publish to the Apple App Store.",
+            "visibleRule": "authType = ApiKey"
+        },
+        {
+            "name": "apiKeyIssuerId",
+            "type": "string",
+            "label": "App Store Connect API Issuer ID",
+            "defaultValue": "",
+            "required": true,
+            "helpMarkDown": "The issuer_id of the account used to publish to the Apple App Store.",
+            "visibleRule": "authType = ApiKey"
+        },
+        {
+            "name": "apiKeyContent",
```

Comment:
@jfahrenkrug Forgot this one: So instead of using a custom key best to use the built-in `apitoken` as otherwise the UI adds a default API Token field that is now not used:
![image](https://user-images.githubusercontent.com/9464097/104325786-aced0c80-54f1-11eb-8091-b8a12820d22e.png)

I think you should change the name to `apitoken` adjust the code to take from this value and leave the label/description as is. This way it will have correct description on the UI (not the default API Token).

Reply:
Simply change the `name` from the current `apiKeyContent` to `apitoken`. This should reuse the built-in field that gets added. Of course you will have to change the usages of `apiKeyContent` and use `apitoken` instead.

Reply:
I see, I'll make that change. Btw, are you a maintainer on this project?

Reply:
No, I'm not :(

Reply:
@egor-bryzgalov What are your thoughts on this? Should we change this as suggested by @PeterStaev?

Reply:
@PeterStaev Could you please clarify why you want this change to be made? I don't see any UI fields being added

![image](https://user-images.githubusercontent.com/72494759/107011589-e6d7c880-67a8-11eb-80a3-d2c35eb7b911.png)

Reply:
@DaniilShmelev , that's because you are looking at it in GitHub actions 😄 In latest version of Azure DevOps Server 2020, it shows an API Token field that is unused:
![image](https://user-images.githubusercontent.com/9464097/107016707-efc79a80-67a6-11eb-9491-b4918b479837.png)

Reply:
Thanks for the great point @PeterStaev. Yeah, we've checked it, this feature is present in the service connection editor.
@jfahrenkrug, I think it makes sense to apply the suggested change.

Reply:
Thanks @PeterStaev and @DaniilShmelev, I've made the change in this commit: https://github.com/microsoft/app-store-vsts-extension/pull/189/commits/2e1cf19b3913ece57a4c8e12348848998f7dacf0 I haven't gotten a chance to actually verify it on an Azure DevOps instance, though. If one of you gets a chance to test it out, that would be great. When you do, could you report back WHERE Azure displays the "Key Content" field now? Before this change, the fields were as follows:
```
[App Store Connect API Key ID]
[App Store Connect API Issuer ID]
[App Store Connect API Key Content (base64-encoded)]
[App Store Connect API In House (Enterprise)]
[API Token]
```

With this change from `apiKeyContent` to `apitoken`, is the new order now (A) or (B)?

## Option A:
```
[App Store Connect API Key ID]
[App Store Connect API Issuer ID]
[App Store Connect API In House (Enterprise)]
[App Store Connect API Key Content (base64-encoded)]
```

## Option B:
```
[App Store Connect API Key ID]
[App Store Connect API Issuer ID]
[App Store Connect API Key Content (base64-encoded)]
[App Store Connect API In House (Enterprise)]
```

I'd prefer option B, because it groups all the important, required fields together on top, instead of listing one of the most important fields ("Key Content") at the end, after the optional "Enterprise" field.

Reply:
I just tested and it is in the "Option B" place. Thanks for doing these changes!
![image](https://user-images.githubusercontent.com/9464097/107051706-71ceb800-67d5-11eb-8429-6f566c9c9ed7.png)

Reply:
🥳 That's great, thanks for testing it Peter.